src/studio/src/designer/backend/Factories/ModelFactory/JsonSchemaToMetamodelConverter.cs

```diff
@@ -806,7 +817,61 @@ private static bool IsRefType(JsonSchema subSchema)
             return refkeyword != null;
         }
 
-        private static bool IsPrimitiveType(JsonSchema subSchema)
+        /// <summary>
+        /// Gets the primitive type flag set, either exlusively ie. it's
+        /// only one flag set or in combination with null ie. it's a
+        /// primitive nullable type.
+        /// </summary>
+        private static SchemaValueType GetPrimitiveType(SchemaValueType type)
```

Comment:
Maybe these methods could be extension methods for SchemaValueType? That way they can be defined in a separate class as public methods. It's my way of sneakily make "utility" classes.

Reply:
Yes, but I'm holding of a bit until I see a broader need in other classes. Right now these are only used within the JsonSchemaToMetamodelConverter class.